chapter2.Rmd

```diff
@@ -285,43 +312,47 @@ names(roulette_vector) <- days_vector
+msg <- paste("&#3629;&#3618;&#3656;&#3634;&#3648;&#3611;&#3621;&#3637;&#3656;&#3618;&#3609;&#3649;&#3611;&#3621;&#3591;&#3605;&#3633;&#3623;&#3649;&#3611;&#3619;", "`poker_vector`, `roulette_vector`", "&#3627;&#3619;&#3639;&#3629;", "`days_vector`")
+# อย่าเปลี่ยนแปลงตัวแปร `poker_vector`, `roulette_vector` หรือ `days_vector`
+
 test_object("poker_vector", undefined_msg = msg, incorrect_msg = msg)
 test_object("roulette_vector", undefined_msg = msg, incorrect_msg = msg)
 test_object("days_vector", undefined_msg = msg, incorrect_msg = msg)
 
-test_object("poker_vector",
-            eq_condition = "equal",
-            incorrect_msg = "Make sure that you assign `days_vector` to the names of `poker_vector`.")
-test_object("roulette_vector",
-            eq_condition = "equal",
-            incorrect_msg = "Make sure that you assign `days_vector` to the names of `roulette_vector`.")
-success_msg("Nice one! A word of advice: try to avoid code duplication at all times. Continue to the next exercise and learn how to do arithmetic with vectors!")
+incorrect_msg <- paste("&#3605;&#3619;&#3623;&#3592;&#3626;&#3629;&#3610;&#3651;&#3627;&#3657;&#3649;&#3609;&#3656;&#3651;&#3592;&#3623;&#3656;&#3634;&#3588;&#3640;&#3603;&#3652;&#3604;&#3657;&#3651;&#3594;&#3657;&#3605;&#3633;&#3623;&#3649;&#3611;&#3619;", "`days_vector`", "&#3651;&#3609;&#3585;&#3634;&#3619;&#3605;&#3633;&#3657;&#3591;&#3594;&#3639;&#3656;&#3629;&#3651;&#3627;&#3657;&#3585;&#3633;&#3610;&#3586;&#3657;&#3629;&#3617;&#3641;&#3621;&#3616;&#3634;&#3618;&#3651;&#3609;")
+# ตรวจสอบให้แน่ใจว่าคุณได้ใช้ตัวแปร `days_vector` ในการตั้งชื่อให้กับข้อมูลภายใน
+
+success_msg <- "&#3648;&#3618;&#3637;&#3656;&#3618;&#3617;&#3652;&#3611;&#3648;&#3621;&#3618;&#33;&#32;&#3592;&#3635;&#3652;&#3623;&#3657;&#3623;&#3656;&#3634;&#3588;&#3640;&#3603;&#3588;&#3623;&#3619;&#3627;&#3621;&#3637;&#3585;&#3648;&#3621;&#3637;&#3656;&#3618;&#3591;&#3585;&#3634;&#3619;&#3614;&#3636;&#3617;&#3614;&#3660;&#3650;&#3588;&#3657;&#3604;&#3648;&#3604;&#3636;&#3617;&#3595;&#3657;&#3635;&#3654;&#3650;&#3604;&#3618;&#3652;&#3617;&#3656;&#3592;&#3635;&#3648;&#3611;&#3655;&#3609;&#32;&#3606;&#3657;&#3634;&#3614;&#3619;&#3657;&#3629;&#3617;&#3649;&#3621;&#3657;&#3623;&#3585;&#3655;&#3652;&#3611;&#3648;&#3619;&#3637;&#3618;&#3609;&#3619;&#3641;&#3657;&#3648;&#3619;&#3639;&#3656;&#3629;&#3591;&#3585;&#3634;&#3619;&#3588;&#3635;&#3609;&#3623;&#3603;&#3650;&#3604;&#3618;&#3651;&#3594;&#3657;&#32;&#118;&#101;&#99;&#116;&#111;&#114;&#32;&#3651;&#3609;&#3649;&#3610;&#3610;&#3613;&#3638;&#3585;&#3627;&#3633;&#3604;&#3605;&#3656;&#3629;&#3652;&#3611;&#3652;&#3604;&#3657;&#3648;&#3621;&#3618;"
+# เยี่ยมไปเลย! จำไว้ว่าคุณควรหลีกเลี่ยงการพิมพ์โค้ดเดิมซ้ำๆโดยไม่จำเป็น ถ้าพร้อมแล้วก็ไปเรียนรู้เรื่องการคำนวณโดยใช้ vector ในแบบฝึกหัดต่อไปได้เลย
+
+test_object("poker_vector", eq_condition = "equal", incorrect_msg = paste(incorrect_msg, "`poker_vector`"))
```

Comment:
I was about to comment on the incorrect message above that you missed the variable name, but then I realized you are too smart 👍